dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -952,102 +1078,143 @@ private void setStructureDependencies() throws DotDataException, DotSecurityExce
-            this.categories.addOrClean(category.getCategoryId(), category.getModDate());
-        });
+		// Categories Dependencies
+		if(!publisherFilter.doesExcludeDependencyClassesContainsType(PusheableAsset.CATEGORY.getType())) {
+			APILocator.getCategoryAPI()
+					.findCategories(new StructureTransformer(structure).from(), user)
+					.forEach(category -> {
+						this.categories.addOrClean(category.getCategoryId(), category.getModDate());
+					});
+		}
 
 		// Related structures
-		final List<Relationship> relations = FactoryLocator.getRelationshipFactory().byContentType(structure);
-
-		for (final Relationship relationship : relations) {
-			relationships.addOrClean( relationship.getInode(), relationship.getModDate());
-
-			if(!structures.contains(relationship.getChildStructureInode()) && config.getOperation().equals( Operation.PUBLISH) ){
-				Structure struct = CacheLocator.getContentTypeCache().getStructureByInode(relationship.getChildStructureInode());
-				solvedStructures.add(stInode);
-				structures.addOrClean( relationship.getChildStructureInode(), struct.getModDate());
-
-				if(!solvedStructures.contains(relationship.getChildStructureInode()))
-					structureDependencyHelper( relationship.getChildStructureInode() );
-			}
-			if(!structures.contains(relationship.getParentStructureInode()) && config.getOperation().equals( Operation.PUBLISH) ){
-				Structure struct = CacheLocator.getContentTypeCache().getStructureByInode(relationship.getParentStructureInode());
-				solvedStructures.add(stInode);
-				structures.addOrClean( relationship.getParentStructureInode(), struct.getModDate());
-
-				if(!solvedStructures.contains(relationship.getParentStructureInode()))
-					structureDependencyHelper( relationship.getParentStructureInode() );
+		if(publisherFilter.isRelationships()) {
+			final List<Relationship> relations = FactoryLocator.getRelationshipFactory()
+					.byContentType(structure);
+
+			for (final Relationship relationship : relations) {
+				relationships.addOrClean(relationship.getInode(), relationship.getModDate());
+
+				if (!contentTypes.contains(relationship.getChildStructureInode()) && config
+						.getOperation().equals(Operation.PUBLISH)) {
+					final Structure childStructure = CacheLocator.getContentTypeCache()
+							.getStructureByInode(relationship.getChildStructureInode());
+					solvedContentTypes.add(stInode);
+					contentTypes
+							.addOrClean(relationship.getChildStructureInode(), childStructure.getModDate());
+
+					if (!solvedContentTypes.contains(relationship.getChildStructureInode())) {
+						structureDependencyHelper(relationship.getChildStructureInode(),
+								publisherFilter);
+					}
+				}
+				if (!contentTypes.contains(relationship.getParentStructureInode()) && config
+						.getOperation().equals(Operation.PUBLISH)) {
+					Structure struct = CacheLocator.getContentTypeCache()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'struct' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)